src/main/scala/Cqrs/BatchRunner.scala

```diff
@@ -4,72 +4,44 @@ import cats.data.Xor
 import cats.state.StateT
 
 import Cqrs.Aggregate.{ AggregateDef, AggregateState, EventDatabaseWithFailure, Error }
-import Cqrs.InMemoryDb.{ DbBackend, runInMemoryDb }
-import lib.{ HList, HNil }
-import lib.HList._
+import Cqrs.DbAdapters.InMemoryDb.{EventSerialisation, DbBackend, runInMemoryDb}
+import cats.~>
 
-object BatchRunner {
-  def empty = BatchRunner[HNil, HNil.type](HNil, HNil)
+import shapeless._
```

Comment:
Still had to leave my hlist with the kmap operation - looks like shapeless has output type specified by operations and couldn't find an easy way to implement this kind of mapping (to the exactly same type, specified by caller). At least it's using shapeless type for the hlist data itself :)